### PR TITLE
MINOR: Improve shutdown sequence

### PR DESCRIPTION
**Pull request by ijuma**

### CODE REVIEW

File: core/src/main/scala/kafka/server/KafkaServer.scala [hunk 1]
```diff
@@ -589,28 +589,35 @@ class KafkaServer(val config: KafkaConfig, time: Time = Time.SYSTEM, threadNameP
...
         if (adminManager != null)
           CoreUtils.swallow(adminManager.shutdown())
+
         if (transactionCoordinator != null)
           CoreUtils.swallow(transactionCoordinator.shutdown())
         if (groupCoordinator != null)
           CoreUtils.swallow(groupCoordinator.shutdown())
+
+        if (replicaManager != null)
+          CoreUtils.swallow(replicaManager.shutdown())
```
Comment (ijuma) [inline core/src/main/scala/kafka/server/KafkaServer.scala:612]: It seems like the ReplicaManager should only be closed after the coordinators who use it (and the LogManager as a result).

File: core/src/main/scala/kafka/server/KafkaServer.scala [hunk 2]
```diff
@@ -589,28 +589,35 @@ class KafkaServer(val config: KafkaConfig, time: Time = Time.SYSTEM, threadNameP
...
         if (logManager != null)
           CoreUtils.swallow(logManager.shutdown())
+
         if (kafkaController != null)
           CoreUtils.swallow(kafkaController.shutdown())
```
Comment (ijuma) [inline core/src/main/scala/kafka/server/KafkaServer.scala:617]: I was a bit unsure about this one. Does it make sense for it to be the last thing to be shutdown?

**Comment by ijuma** [comment]
@junrao, this is a tiny change, but I also left one question.

**Comment by ijuma** [comment]
retest this please

File: core/src/main/scala/kafka/server/KafkaServer.scala (hunk 2 above)
Comment (junrao) [reply core/src/main/scala/kafka/server/KafkaServer.scala:617]: It probably can be done early. Not sure if it hurts to shutdown it here.

**Comment by junrao** [comment]
@ijuma : Thanks for the patch. LGTM

File: core/src/main/scala/kafka/server/KafkaServer.scala (hunk 2 above)
Comment (ijuma) [reply core/src/main/scala/kafka/server/KafkaServer.scala:617]: OK, let's leave it like that then.

**Pull request closed without merging**